UI/GovernmentWnd.cpp

```diff
@@ -0,0 +1,1538 @@
+
+    int empire_id = HumanClientApp::GetApp()->EmpireID();
+    const Empire* empire = GetEmpire(empire_id);  // may be nullptr
+
+    int cur_col = NUM_COLUMNS;
+    std::shared_ptr<PoliciesListBoxRow> cur_row;
+    int num_policies = 0;
+
+    // remove policies currently in rows of listbox
+    Clear();
+
+    /// filter policies by availability and current designation of categories
+    // for display
+    auto cats_policies = GroupAvailableDisplayablePolicies(empire);
+
+    // get empire id and location to use for cost and time comparisons
+    int loc_id = INVALID_OBJECT_ID;
+    if (empire) {
+        auto location = Objects().get(empire->CapitalID());
+        loc_id = location ? location->ID() : INVALID_OBJECT_ID;
+    }
+
+    for (auto& cat : cats_policies) {
+        //std::cout << "  cat: " << cat.first << std::endl;
+        // take the sorted policies and make UI element rows for the PoliciesListBox
+        for (const auto* policy: cat.second) {
+            //std::cout << "   ... policy: " << policy->Name() << std::endl;
+            // check if current row is full, and make a new row if necessary
+            if (cur_col >= NUM_COLUMNS) {
+                if (cur_row)
+                    Insert(cur_row);
+                cur_col = 0;
+                cur_row = GG::Wnd::Create<PoliciesListBoxRow>(
+                    TOTAL_WIDTH, SLOT_CONTROL_HEIGHT + GG::Y(PAD), m_availabilities_state);
+            }
+            ++cur_col;
+            ++num_policies;
+
+            // make new policy control and add to row
+            auto control = GG::Wnd::Create<PolicyControl>(policy);
+            control->ClickedSignal.connect(PoliciesListBox::PolicyClickedSignal);
+            control->DoubleClickedSignal.connect(PoliciesListBox::PolicyDoubleClickedSignal);
+            control->RightClickedSignal.connect(PoliciesListBox::PolicyRightClickedSignal);
+
+            cur_row->push_back(control);
+
+            //std::cout << "PoliciesListBox::Populate created row for policy: " << policy->Name() << std::endl;
+        }
+    }
+    // add any incomplete rows
+    if (cur_row)
+        Insert(cur_row);
+
+    // keep track of how many columns are present now
+    m_previous_num_columns = NUM_COLUMNS;
+
+    // If there are no policies add a prompt to suggest a solution.
+    if (num_policies == 0)
+        Insert(GG::Wnd::Create<PromptRow>(TOTAL_WIDTH,
+                                          UserString("ALL_AVAILABILITY_FILTERS_BLOCKING_PROMPT")),
+               begin(), false);
+}
+
+void PoliciesListBox::ShowCategory(const std::string& category, bool refresh_list) {
+    if (m_policy_categories_shown.find(category) == m_policy_categories_shown.end()) {
+        m_policy_categories_shown.insert(category);
+        if (refresh_list)
+            Populate();
+    }
+}
+
+void PoliciesListBox::ShowAllCategories(bool refresh_list) {
+    auto cats = GetPolicyManager().PolicyCategories();
+    for (const auto& category : cats)
```

Comment:
```suggestion
    m_policy_categories.insert(cats.begin(), cats.end());
```

instead of the loop.